xlive/Blam/Engine/main/main_game.cpp

```diff
@@ -0,0 +1,213 @@
+#include "stdafx.h"
+#include "main_game.h"
+
+#include "Blam/Engine/cache/cache_files.h"
+#include "Blam/Engine/cseries/cseries.h"
+#include "Blam/Engine/cseries/cseries_strings.h"
+#include "Blam/Engine/game/game.h"
+#include "Blam/Engine/Networking/Session/NetworkSession.h"
+#include "Blam/Engine/saved_games/game_variant.h"
+
+#include "Util/Hooks/Hook.h"
+
+s_game_options g_main_game_launch_options;
+DWORD g_main_game_launch_user_count;
+
+// Setup default values for the options structure depending on the game mode set
+void main_game_launch_setup_game_mode_details(void);
+
+void main_game_initialize()
+{
+    game_options_new(&g_main_game_launch_options);
+    return;
+}
+
+void main_game_launch_set_map_name(const char* map_name)
+{
+    MultiByteToWideChar(CP_UTF8, 0, map_name, -1, g_main_game_launch_options.scenario_path.get_string(), 260);
+    return;
+}
+
+// TODO rewrite the obfuscated function
+void main_game_change(const s_game_options* options)
+{
+    typedef void(__stdcall* main_game_change_t)(const s_game_options*);
+    auto p_main_game_change = Memory::GetAddress<main_game_change_t>(0x89BA, 0x1E4EC);
+    p_main_game_change(options);
+    return;
+}
+
+void main_game_launch_set_difficulty(short difficulty)
+{
+    if (!VALID_INDEX(difficulty, k_campaign_difficulty_levels_count))
+    {
```

Comment:
Just return here if validation fails and remove the else statement. Also do it for the other similar cases